front-end/src/components/IntentDiagram.test.js

```diff
@@ -0,0 +1,69 @@
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import IntentDiagram from './IntentDiagram.jsx';
+
+describe("IntentDiagram", () => {
+    let props;
+    const renderComponent = (props) => {
+        render(<IntentDiagram {...props} />)
+    }
+    beforeEach(() => {
+        props = {
+            question: "question",
+            branches: {
+                q1: 100
+            },
+        };
+    });
+    it("correctly displays the question", () => {
+        renderComponent(props);
+        const question = screen.getByTestId(props.question);
+        expect(question).toBeInTheDocument();
+        expect(question.textContent).toBe(props.question);
+    });
+    it("correctly displays 1 branch", () => {
+        renderComponent(props);
+        const q1 = screen.getByTestId("q1");
+        const percentage = screen.getByTestId("q1-100");
+        expect(q1).toBeInTheDocument();
+        expect(q1.textContent).toBe("q1");
+        expect(percentage).toBeInTheDocument();
+        expect(percentage.textContent).toBe(`${props.branches.q1}%`)
+    });
+
+    it("correctly displays 2 branches", () => {
+        props.branches = {
+            q1: 50,
+            q2: 50,
+        };
+        renderComponent(props);
+        for (const key of Object.keys(props.branches)) {
+        const value = props.branches[key];
+        const q = screen.getByTestId(key);
+        const percentage = screen.getByTestId(`${key}-${value}`);
+        expect(q).toBeInTheDocument();
+        expect(q.textContent).toBe(key);
+        expect(percentage).toBeInTheDocument();
+        expect(percentage.textContent).toBe(`${value}%`)
+        }
+    });
+
+    it("correctly displays 3 branches", () => {
+        props.branches = {
+            q1: 50,
+            q2: 20,
+            q3: 30,
```

Comment:
This is more a general question related to testing, would there be any additional value in randomizing these percentages instead of hard-coding them into the test? I've seen some testing scripts do that, especially if the test isn't testing a specific edge case that requires specific values. It shouldn't make a difference for a simpler app like ours, just curious.

Reply:
Randomizing could make sense if testing if a specific algorithm is applied correctly to them (i.e. we randomize the percentages, divide by 2, and then test to see if that is what is being displayed), but since this is 1 for 1, I think using hard-coded values should be fine. 

Essentially, I think both random and non-random values would be valid in this case.